pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -65,8 +70,58 @@ func setAccessLog(env *model.Environment, node *model.Proxy, config *tcp_proxy.T
 			acc.ConfigType = &accesslog.AccessLog_Config{Config: util.MessageToStruct(fl)}
 		}
 
-		config.AccessLog = []*accesslog.AccessLog{acc}
+		config.AccessLog = append(config.AccessLog, acc)
+
+	}
+
+	if env.Mesh.EnvoyAccesslogService != nil && env.Mesh.EnvoyAccesslogService.TargetUri != "" {
+		googleGrpc := &core.GrpcService_GoogleGrpc{
+			TargetUri:  env.Mesh.EnvoyAccesslogService.TargetUri,
+			StatPrefix: tcpEnvoyAccesslogName,
+		}
+		if env.Mesh.EnvoyAccesslogService.Credentials != nil {
+			c := env.Mesh.EnvoyAccesslogService.Credentials
+			sslCred := &core.GrpcService_GoogleGrpc_SslCredentials{}
+			isValid := false
+			if c.RootCerts != "" {
+				sslCred.RootCerts = &core.DataSource{Specifier: &core.DataSource_Filename{Filename: c.RootCerts}}
+				isValid = true
+			}
+			if c.PrivateKey != "" && c.CertChain != "" {
+				sslCred.PrivateKey = &core.DataSource{Specifier: &core.DataSource_Filename{Filename: c.PrivateKey}}
+				sslCred.CertChain = &core.DataSource{Specifier: &core.DataSource_Filename{Filename: c.CertChain}}
+				isValid = true
+			}
+			if isValid {
+				googleGrpc.ChannelCredentials = &core.GrpcService_GoogleGrpc_ChannelCredentials{
+					CredentialSpecifier: &core.GrpcService_GoogleGrpc_ChannelCredentials_SslCredentials{
+						SslCredentials: sslCred,
+					},
+				}
+			}
+		}
+		fl := &accesslogconfig.TcpGrpcAccessLogConfig{
+			CommonConfig: &accesslogconfig.CommonGrpcAccessLogConfig{
+				LogName: tcpEnvoyAccesslogName,
+				GrpcService: &core.GrpcService{
+					TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+						GoogleGrpc: googleGrpc,
```

Comment:
Why does this only support googlegrpc? Why not envoygrpc as well ?

Reply:
So the envoy grpc means we have to have a service inside the k8s cluster, so that there is an envoy cluster for that service. Since these are access logs (large volume of data), they tend to go to hosted entities like stack driver, hosted elasticsearch on aws, etc.. If we use envoy grpc, then we have to have a service entry for such things, ensure that this service entry is visible to ALL namespaces and gets inherited via sidecar, encode destination rules with TLS settings for such a service entry, so on so forth, just to add this Envoy cluster with the right set of TLS settings. Thats a lot of configs, and management overhead with lots of inconsistency issues, compared to this single shot configuration that the access log filter uses to directly communicate with the service. I believe the TD team in google is using this a lot as well (it came from them IIRC).

Reply:
I asked this elsewhere, but why did that not also apply for the metrics service?  Is the expectation that the metrics service will always be in-cluster?  I'm mostly interested in consistency between the way the two telemetry services are exposed via Istio configuration.

Reply:
Google grpc satisfies stackdriver and other similar use cases.
However daemon set with  uds upstream is also a valid scenario.

Reply:
> I asked this elsewhere, but why did that not also apply for the metrics service? Is the expectation that the metrics service will always be in-cluster? I'm mostly interested in consistency between the way the two telemetry services are exposed via Istio configuration.

It absolutely applies for the metrics service as well. I didn't know that google grpc existed in envoy and that it was supported for metrics. Had I known it in advance, I would have asked the salesforce people to add google grpc support as well. 

> However daemon set with uds upstream is also a valid scenario.

Yep. But trying to do this piece meal. Just to get to this, I had to go through a lot of FUD about touching global configs, and this PR was left hanging for 30+ days :).

Reply:
> It absolutely applies for the metrics service as well. I didn't know that google grpc existed in envoy and that it was supported for metrics. Had I known it in advance, I would have asked the salesforce people to add google grpc support as well.

OK. Sounds good. Let's work together on bringing some consistency for both of these.

Reply:
google gRPC library does not scale to data path traffic. Access logs seem like comparable to data plane traffic in bandwidth, so it seems inappropriate to use gRPC lib for ALS.

Reply:
If that is the case, why does envoy access log have support for both google grpc and envoy grpc?

Reply:
It supports Google gRPC library for limited, control-plane related scale. I don't think Envoy recommends using Google gRPC for data plane at all.

Reply:
@kyessenov position is correct. We're working with gRPC team (CC @markdroth) to improve this situation, but this is a long term project.

Reply:
@douglas-reid I moved to envoy grpc.. consistent with metrics